low-level/files/opt/seagate/sspl/setup/health_provider/storage.py

```diff
@@ -0,0 +1,153 @@
+#!/bin/python3
+
+# CORTX Python common library.
+# Copyright (c) 2021 Seagate Technology LLC and/or its Affiliates
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com
+
+"""
+ ***************************************************************************
+  Description: Storage class to provide fru health and specific information
+               through controller web service.
+ ***************************************************************************
+"""
+
+import errno
+import json
+import re
+import time
+
+from framework.utils.conf_utils import (
+    GLOBAL_CONF, SRVNODE, Conf, STORAGE_TYPE_KEY, NODE_TYPE_KEY, SITE_ID_KEY,
+    RACK_ID_KEY, NODE_ID_KEY, CLUSTER_ID_KEY, RELEASE, TARGET_BUILD)
+from error import HealthProviderError
+from resource_health import ResourceHealth
+
+
+class StorageHealth(ResourceHealth):
+    """Provides health information of FRUs in storage"""
+
+    name = "storage"
+
+    def __init__(self):
+        """Initialize storage"""
+        super().__init__()
+        self.validate_storage_type_support()
+        self.storage_frus = {
+            "controllers": self.get_controllers_info
+            }
+
+    @staticmethod
+    def validate_storage_type_support():
+        """Check for supported storage type"""
+        storage_type = Conf.get(GLOBAL_CONF, STORAGE_TYPE_KEY, "virtual").lower()
+        supportted_types = ["5u84", "rbod", "pods"]
```

Comment:
typo -> supported_types

pods official name is corvault, so lets add this too to the list, in addition to pods

Reply:
Done.